ctcf_orientation.py

```diff
@@ -7,79 +7,150 @@
-                    loop_anchor_2_list.append(ctcf_row['strand'])
-        # Now we have the CTCF orientations (strands) of each anchor in the loop, check and see if they are
-        # First convergent, then tandem, then divergent, then single, then none
-        # First check if either list is empty, automatically none
-        # If the first anchor has no CTCF motifs (if empty list):
-        # print('Printing anchor 1 list', loop_anchor_1_list)
-        # print('Printing anchor 2 list', loop_anchor_2_list)
-        if not loop_anchor_1_list:
-            # Check if second anchor is empty also, if so then the loop is categorized as none
-            if not loop_anchor_2_list:
-                none += 1
-            # If the second anchor is not empty, then the loop has a single CTCF motif
-            if loop_anchor_2_list:
-                single += 1
-        # if the second anchor list is empty, already counted if they are both empty
+# Initialize the counts as global variables so multiple methods can access them
+convergent = 0
+tandem = 0
+divergent = 0
+single = 0
+none = 0
+
+
+def main():
+    # parse command line arguments for loop, motif, and output files
+    loopfile, motiffile, outputfile = initialize()
+
+    # read in loop dataframe
+    loop_df = read_loop(loopfile)
+
+    # read in motif dataframe
+    motif_df = read_motif(motiffile)
+
+    # loop through each row in the loop dataframe and update the counts of each kind of motif
+    loop_df.apply(lambda x: process_loop_row(motif_df, x), axis=1)
+    ctcf_orientation_counts = [convergent, tandem, divergent, single, none]
+
+    # plot and display the output of the orientation counts
+    plot_output(ctcf_orientation_counts, outputfile)
+
+
+# searches the ctcf motifs dataframe for overlaps with the inputted values
+def find_strand(ctcf_df, loop_chrom, loop_row, strand_num):
+    # subset the ctcf dataframe to only contain the correct chromosome for this row
+    ctcf_chr = ctcf_df.loc[(ctcf_df['chrom'] == loop_chrom)]
+    # if we're looking at anchor list 1...
+    if strand_num == 1:
+        # return the strand values for every location in the ctcf dataframe subset which contains EITHER:
+        # a CTCF start site between the first start and end boundaries of the loop row OR
+        # a CTCF end site betwen the first start and end boundaries of the loop row
+        return ctcf_chr.loc[((ctcf_chr['start'] >= loop_row['start1']) & (ctcf_chr['start'] <= loop_row['end1'])) |
+                            ((ctcf_chr['end'] >= loop_row['start1']) &
+                             (ctcf_chr['end'] <= loop_row['end1'])), 'strand'].tolist()
+    # same thing as above for anchor list 2,
+    return ctcf_chr.loc[((ctcf_chr['start'] >= loop_row['start2']) & (ctcf_chr['start'] <= loop_row['end2'])) |
+                        ((ctcf_chr['end'] >= loop_row['start2']) &
+                         (ctcf_chr['end'] <= loop_row['end2'])), 'strand'].tolist()
```

Comment:
Since CTCF sites are very small (around 20bp) I like the idea of having the start and end of the CTCF site within the bounds of the loop, since for them to be functional the loop should have the entire CTCF site.. something like what I had earlier with having the CTCF sites within the range?